doc/rst/developer/bestPractices/buildingdocs.rst

```diff
@@ -1,15 +1,227 @@
+extensions for editors to highlight Chapel code.
+
+The .rst files are also text.  The Chapel files that are being converted into .rst
+files have rst text in comments.  There are some previewers for .rst text in
+vscode and atom, but none of them are quite as useful as editing a .rst file
+in Github and using the preview option.  The way to see all of the links that
+are created is to do the ``make docs`` that is described above.
+
+
+Files that live in a repository
+===============================
+Some of the documentation files are not converted into html files.  Instead
+they are raw html files in the chapel-www repository (currently only accessible
+by core developers) or raw .rst files that live in the public Github respository.
+
+Examples include:
+ - https://chapel-lang.org/contributing.html
+ - https://github.com/chapel-lang/chapel/blob/main/doc/rst/developer/bestPractices/ContributorInfo.rst
+
+
+
+Linking between files
+=======================================
+
+Linking between files when both files are in https://chapel-lang.org/docs/.
+ - RST files in the `docs/rst/` or `chapel/test/release/examples/primers/`
+   subdirectory trees can refer to each other and
+   when they are converted to html, those are converted to links.
+ - The file being linked to has a ``.. _file-tag-name:`` at the top of it.
+ - The file linking to another file uses ``:ref:`file-tag-name``` anywhere
+   in the text.
+ - See https://github.com/chapel-lang/chapel/blob/main/test/release/examples/primers/interopWithC.chpl
+   for an example of ``:ref:`readme-libraries``` that links to
+   the html for
+   https://raw.githubusercontent.com/chapel-lang/chapel/main/doc/rst/technotes/libraries.rst. The rst source file has
+   ``.. _readme-libraries:`` at the top.
+
+
+Linking between files in https://chapel-lang.org/docs/ and
+html files in https://chapel-lang.org .
+
+- Linking from an rst file to an html file directly in https://chapel-lang.org
+  is done by naming the needed URL in the .rst file and using that name.
+  Here are some examples:
+
+``.. _Chapel Developers: https://chapel.discourse.group/c/developers``
+
+``.. _Chapel Users: https://chapel.discourse.group/c/users``
+
+``.. _chapel-lang/chapel: https://github.com/chapel-lang/chapel``
+
+Here is a reference to the ```Chapel Developers`_`` URL later in the .rst file.
+
+- Linking from one of the html files at https://chapel-lang.org such as
+  https://chapel-lang.org/contributing.html can only be done by a core contributor.
+  Post an inquiry to the `Chapel Developers`_ Discourse forum.
+
+
+============================================
+Organization of Chapel Documentation Sources
+============================================
+
+The file chapel/doc/rst/index.rst is the root file that sets up the
+structure of the documentation and results in the current sidebar
+displayed on https://chapel-lang.org/docs/.
+
+
+There is a strong correlation in side bar of
```

Comment:
"in side bar" -> "between the sidebar" (or "side bar")

Reply:
done